tasks/pkg-suse.yml

```diff
@@ -24,31 +24,30 @@
     state: present
   when: not ansible_check_mode
 
+- set_fact:
+   zypper_repo_url: "{% if datadog_agent5 %}{{datadog_agent5_zypper_repo}}{% else %}{{datadog_zypper_repo}}{% endif %}"
```

Comment:
any reason not to put this logic directly in the template? Or alternatively, set the variable in the `vars` of the next task.